### PR TITLE
feat: Add owner references to all resources

### LINKED ISSUE #498: Add ownerReference to managed entities
If I delete a database, resources created by it (services, statefulsets, etc.) should be deleted automatically. This can be easily implemented by adding the Postgres resource as an `ownerReference` (with `controller: true`). Kubernetes will then automatically delete the resources once the master database resource is gone.

### PR DESCRIPTION
**Pull request by mbegenau**
Previously, only the Service and Deployment resource managed by the operator had an OwnerReference linking it to the StatefulSet. Now, all managed resources get an OwnerReference pointing to the Postgresql resource.

Fixes #498.

### CODE REVIEW

**Comment by batazor** [comment]
It would be cool to add support for OwnerReferences references. Is there anything that prevents us from adopting this PR?

**Comment by sass1997** [comment]
+1

**Comment by jhpinson** [comment]
+1

**Comment by Yingrjimsch** [comment]
+1

**Comment by amitde69** [comment]
+1

**Comment by metajiji** [comment]
+1 any news?

**Comment by Demch1k** [comment]
Hello everyone!
When we wil see it in postgres operator?
It's well needed feature for us, because we can't use ArgoCD with zalando operator without owner references.

**Comment by fm2022aa** [comment]
+1

File: pkg/cluster/k8sres.go [hunk 1]
```diff
@@ -2065,6 +2068,7 @@ func (c *Cluster) generateEndpoint(role PostgresRole, subsets []v1.EndpointSubse
 			Namespace:   c.Namespace,
 			Annotations: c.annotationsSet(nil),
 			Labels:      c.roleLabelsSet(true, role),
+      OwnerReferences: c.ownerReferences(),
```
Comment (FxKu) [inline pkg/cluster/k8sres.go:2071]:
```suggestion
			OwnerReferences: c.ownerReferences(),
```

File: pkg/cluster/k8sres_test.go [hunk 2]
```diff
@@ -2320,104 +2320,128 @@ func TestSidecars(t *testing.T) {
...
+
+	testLabelsAndSelectors := func(cluster *Cluster, podDisruptionBudget *policyv1.PodDisruptionBudget) error {
+		if podDisruptionBudget.ObjectMeta.Namespace != "myapp" {
+			return fmt.Errorf("Object Namespace incorrect.")
+		}
+		if !reflect.DeepEqual(podDisruptionBudget.Labels, map[string]string{"team": "myapp", "cluster-name": "myapp-database"}) {
+
+			return fmt.Errorf("Labels incorrect.")
+		}
+		if !reflect.DeepEqual(podDisruptionBudget.Spec.Selector, &metav1.LabelSelector{
```
Comment (FxKu) [inline pkg/cluster/k8sres_test.go:2354]:
```suggestion
		masterLabelSelectorDisabled := cluster.OpConfig.PDBMasterLabelSelector != nil && !*cluster.OpConfig.PDBMasterLabelSelector

		if podDisruptionBudget.ObjectMeta.Namespace != "myapp" {
			return fmt.Errorf("Object Namespace incorrect.")
		}
		if !reflect.DeepEqual(podDisruptionBudget.Labels, map[string]string{"team": "myapp", "cluster-name": "myapp-database"}) {

			return fmt.Errorf("Labels incorrect.")
		}
		if !masterLabelSelectorDisabled &&
			!reflect.DeepEqual((podDisruptionBudget.Spec.Selector, &metav1.LabelSelector{
```

Comment (FxKu) [reply pkg/cluster/k8sres_test.go:2354]: And final test case must be changed to
```
		{
			scenario: "With PDBMasterLabelSelector disabled",
			spec: New(
				Config{OpConfig: config.Config{Resources: config.Resources{ClusterNameLabel: "cluster-name", PodRoleLabel: "spilo-role"}, PDBNameFormat: "postgres-{cluster}-pdb", EnablePodDisruptionBudget: util.True(), PDBMasterLabelSelector: util.False()}},
				k8sutil.KubernetesClient{},
				acidv1.Postgresql{
					ObjectMeta: metav1.ObjectMeta{Name: "myapp-database", Namespace: "myapp"},
					Spec:       acidv1.PostgresSpec{TeamID: "myapp", NumberOfInstances: 3}},
				logger,
				eventRecorder),
			check: []func(cluster *Cluster, podDisruptionBudget *policyv1.PodDisruptionBudget) error{
				testPodDisruptionBudgetOwnerReference,
				hasName("postgres-myapp-database-pdb"),
				hasMinAvailable(1),
				testLabelsAndSelectors,
			},
		},
```

**Pull request closed without merging**